lib/handlers/error-pages.js

```diff
@@ -163,15 +163,12 @@ function redirectToSelectProvider (req, res) {
   res.status(401)
   res.header('Content-Type', 'text/html')
 
-  let currentUrl = util.fullUrlForReq(req)
-  req.session.returnToUrl = currentUrl
```

Comment:
We can't rely on session, because the server cannot see the hash.

Reply:
This fixed mechanism (not relying on session) will still work on direct requests by the browser (like, requesting an image or a pdf file or whatever), not mediated by a client?

Reply:
If we end up in this code, we are in the process of rendering an HTML error page. So if that's the behavior we exhibit in response to a browser directly requesting an image (I hope not), it's broken for another reason already :smile:

Reply:
Ahh, I see. Makes sense.